tests/vxlan/vxlan_ecmp_utils.py

```diff
@@ -0,0 +1,599 @@
+            if type in element['name']:
+                if ipaddress.ip_address(element['addr']).version == 4:
+                    all_neighbors[element['name']] = {"v4":element['addr']}
+                elif ipaddress.ip_address(element['addr']).version == 6:
+                    all_neighbors[element['name']].update({"v6":element['addr']})
+        return all_neighbors
+
+    def get_corresponding_ports(self, shut_intf_list, minigraph_data):
+        '''
+        This is for tests that shutdown some of the T2 ports.
+        This function will check which ports are to be ignored for the encap packets coming
+        back to the PTF. If the encap packet comes in any of these ports, it is a bug.
+        '''
+        eth_ifaces_list = []
+        for intf in shut_intf_list:
+            if "Ethernet" in intf:
+                eth_ifaces_list.append(intf)
+            elif "PortChannel" in intf:
+                for port in self.get_ethernet_ports([intf], minigraph_data):
+                    eth_ifaces_list.append(port)
+        return_list = [minigraph_data["minigraph_ptf_indices"][iface] for iface in eth_ifaces_list]
+        return return_list
+
+    def get_ethernet_ports(self, intf_list, minigraph_data):
+        '''
+            The given interface list can be either Ethernet or Portchannel.
+            This function will return a flat list of Ethernet ports corresponding to
+            the given intf_list itself, or members of Portchannels.
+        '''
+        ret_list = []
+        for intf in intf_list:
+            if "Ethernet" in intf:
+                ret_list.append(intf)
+            elif "PortChannel" in intf:
+                ret_list.extend(minigraph_data['minigraph_portchannels'][intf]['members'])
+
+        return ret_list
+    
+    def gather_ptf_indices_t2_neighbor(self, minigraph_facts, all_t2_neighbors, t2_neighbor, encap_type):
+         # All T2 Neighbors VM's name to Neighbor IP Mapping
+        all_pcs = minigraph_facts['minigraph_portchannel_interfaces']
+        #Neighbor IP to Portchannel interfaces mapping
+        pc_to_ip_map = {}
+        for each_pc in all_pcs:
+            pc_to_ip_map[each_pc['peer_addr']] = each_pc['attachto']
+        #Finding the portchannel under shutdown T2 Neighbor
+        required_pc = pc_to_ip_map[all_t2_neighbors[t2_neighbor][self.get_outer_layer_version(encap_type)].lower()]
+        #Finding ethernet interfaces under that specific portchannel
+        required_ethernet_interfaces = minigraph_facts['minigraph_portchannels'][required_pc]['members']
+        #Finding interfaces with PTF indices
+        ret_list = []
+        for iface in required_ethernet_interfaces:
+            ret_list.append(minigraph_facts["minigraph_ptf_indices"][iface])
+        return ret_list
+
+    def ptf_config(self, duthost, ptfhost, tbinfo, delete_member_a1 = None, delete_member_a2 = None):
+        responder_output = ptfhost.command('supervisorctl stop bfd_responder')
```

Comment:
When bfd_responder is not running on PTF, we're getting an error here.
`bfd_responder: ERROR (no such process)`
Solution:
```suggestion
        responder_output = ptfhost.command('supervisorctl stop bfd_responder', module_ignore_errors=True)
```